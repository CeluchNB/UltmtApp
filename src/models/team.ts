import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { LocalUser, Team } from '../types/team'

export const DisplayTeamSchema: Realm.ObjectSchema = {
    name: 'DisplayTeam',
    embedded: true,
    properties: {
        _id: 'string',
        place: 'string',
        name: 'string',
        teamname: 'string',
        seasonStart: 'string',
        seasonEnd: 'string',
    },
}

export const GuestTeamSchema: Realm.ObjectSchema = {
    name: 'GuestTeam',
    embedded: true,
    properties: {
        _id: 'string?',
        place: 'string?',
        name: 'string',
        teamname: 'string?',
        seasonStart: 'string?',
        seasonEnd: 'string?',
    },
}

export class TeamSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Team',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            place: 'string',
            name: 'string',
            teamname: 'string',
            managers: { type: 'list', objectType: 'DisplayUser' },
            players: { type: 'list', objectType: 'DisplayUser' },
            seasonStart: 'string',
            seasonEnd: 'string',
            seasonNumber: 'int',
            continuationId: 'string',
            rosterOpen: 'bool',
        },
    }

    _id: string
    place: string
    name: string
    teamname: string
    managers: DisplayUser[]
    players: LocalUser[]
    seasonStart: string
    seasonEnd: string
    seasonNumber: number
    continuationId: string
    rosterOpen: boolean

    constructor(team: Team) {
        this._id = team._id
        this.place = team.place
        this.name = team.name
        this.teamname = team.teamname
        this.managers = team.managers
        this.players = team.players
        this.seasonStart = team.seasonStart
        this.seasonEnd = team.seasonEnd
        this.seasonNumber = team.seasonNumber
        this.continuationId = team.continuationId
        this.rosterOpen = team.rosterOpen
    }
}
