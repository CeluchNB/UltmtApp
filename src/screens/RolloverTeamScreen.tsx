import * as React from 'react'
import CheckBox from '@react-native-community/checkbox'
import { Picker } from '@react-native-picker/picker'
import PrimaryButton from '../components/atoms/PrimaryButton'
import { RolloverTeamProps } from '../types/navigation'
import ScreenTitle from '../components/atoms/ScreenTitle'
import { useColors } from '../hooks'
import { Controller, useForm } from 'react-hook-form'
import { StyleSheet, Text, View } from 'react-native'

interface RolloverTeamFormData {
    copyPlayers: boolean
    season: string
}

const RolloverTeamScreen: React.FC<RolloverTeamProps> = ({ route }) => {
    const { colors } = useColors()
    const { id } = route.params
    const { control, handleSubmit } = useForm({
        defaultValues: {
            copyPlayers: true,
            season: '',
        },
    })
    const currentYear = new Date().getFullYear()
    const years = [
        currentYear.toString(),
        `${currentYear} - ${currentYear + 1}`,
        (currentYear + 1).toString(),
    ]

    const styles = StyleSheet.create({
        screen: {
            backgroundColor: colors.primary,
            height: '100%',
            alignItems: 'center',
        },
        checkboxContainer: {
            flexDirection: 'row',
        },
        checkbox: {
            color: colors.textPrimary,
        },
        picker: {
            color: colors.textPrimary,
            width: '75%',
            alignSelf: 'center',
            backgroundColor: colors.primary,
        },
    })

    const rollover = async (data: RolloverTeamFormData) => {
        console.log('rolling over with data', id, data)
    }

    return (
        <View style={styles.screen}>
            <ScreenTitle title="Start New Season" />
            <View style={styles.checkboxContainer}>
                <Text>Keep Current Players</Text>
                <Controller
                    name="copyPlayers"
                    control={control}
                    rules={{ required: true }}
                    render={({ field: { onChange, value } }) => (
                        <CheckBox
                            value={value}
                            onValueChange={onChange}
                            tintColors={{
                                true: colors.textPrimary,
                                false: colors.gray,
                            }}
                            onFillColor={colors.textPrimary}
                            onCheckColor={colors.textPrimary}
                        />
                    )}
                />
            </View>
            <Text>Season</Text>
            <Controller
                name="season"
                control={control}
                rules={{ required: true }}
                render={({ field: { onChange, value } }) => (
                    <Picker
                        style={styles.picker}
                        selectedValue={value}
                        prompt="Season"
                        onValueChange={onChange}>
                        {years.map(year => {
                            return (
                                <Picker.Item
                                    value={year}
                                    label={year}
                                    key={year}
                                />
                            )
                        })}
                    </Picker>
                )}
            />
            <PrimaryButton
                text="Submit"
                loading={false}
                onPress={handleSubmit(rollover)}
            />
        </View>
    )
}

export default RolloverTeamScreen
