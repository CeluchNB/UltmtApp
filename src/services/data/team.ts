import * as Constants from '../../utils/constants'
import { ApiError } from '../../types/services'
import { throwApiError } from '../../utils/service-utils'
import { CreateTeam, Team } from '../../types/team'
import {
    createTeam as networkCreateTeam,
    getManagedTeam as networkGetManagedTeam,
    getTeam as networkGetTeam,
    removePlayer as networkRemovePlayer,
    rollover as networkRollover,
    searchTeam as networkSearchTeam,
    toggleRosterStatus as networkToggleRosterStatus,
} from '../network/team'

/**
 * Method to create team
 * @param token token of user creating team
 * @param data create team data
 * @returns created team
 * @throws error if team is not created successfully
 */
export const createTeam = async (
    token: string,
    data: CreateTeam,
): Promise<Team> => {
    try {
        const response = await networkCreateTeam(token, data)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.CREATE_TEAM_ERROR)
    }
}

/**
 * Method to search for teams by place, name, or username
 * @param term search term
 * @returns list of teams matching term
 * @throws error if backend returns an error
 */
export const searchTeam = async (term: string): Promise<Team[]> => {
    try {
        if (term.length < 3) {
            throw new ApiError('Not enough characters to search.')
        }
        const response = await networkSearchTeam(term)
        return response.data
    } catch (error) {
        return throwApiError(error, Constants.SEARCH_ERROR)
    }
}

/**
 * Method to get team that the current user manages
 * @param token jwt of current user
 * @param id team id
 * @returns team object
 * @throws error if backend returns error
 */
export const getManagedTeam = async (
    token: string,
    id: string,
): Promise<Team> => {
    try {
        const response = await networkGetManagedTeam(token, id)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.GET_TEAM_ERROR)
    }
}

/**
 * Method to get a public team
 * @param id id of team to get
 * @returns team object
 * @throws error if backend returns an error
 */
export const getTeam = async (id: string): Promise<Team> => {
    try {
        const response = await networkGetTeam(id)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.GET_TEAM_ERROR)
    }
}

/**
 * Method to change a team's roster status
 * @param token jwt of user changing team
 * @param id id of team to update
 * @param open boolean for team's roster status
 * @returns updated team object
 * @throws error if backend returns error
 */
export const toggleRosterStatus = async (
    token: string,
    id: string,
    open: boolean,
): Promise<Team> => {
    try {
        const response = await networkToggleRosterStatus(token, id, open)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}

/**
 * Method to remove a player from a team
 * @param token token of current user
 * @param teamId team player is on
 * @param userId id of player to remove
 * @returns updated team object
 * @throws error if backend returns an error
 */
export const removePlayer = async (
    token: string,
    teamId: string,
    userId: string,
): Promise<Team> => {
    try {
        const response = await networkRemovePlayer(token, teamId, userId)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}

/**
 * Method to start a new season for a team. After this method is called,
 * the old team is no longer editable.
 * @param token jwt of manager of team
 * @param teamId id of team to move over
 * @param copyPlayers boolean for deciding if current roster should be kept
 * @param seasonStart start year for season
 * @param seasonEnd start end for season
 * @returns newly created team object
 * @throws error if backend returns an error
 */
export const rollover = async (
    token: string,
    teamId: string,
    copyPlayers: boolean,
    seasonStart: string,
    seasonEnd: string,
): Promise<Team> => {
    try {
        const response = await networkRollover(
            token,
            teamId,
            copyPlayers,
            seasonStart,
            seasonEnd,
        )
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}
