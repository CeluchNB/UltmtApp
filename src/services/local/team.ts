import { Realm } from '@realm/react'
import { Team } from '../../types/team'
import { TeamSchema } from '../../models'
import { getRealm } from '../../models/realm'

export const saveTeams = async (teams: Team[]) => {
    const realm = await getRealm()
    realm.write(() => {
        for (const team of teams) {
            realm.create('Team', team, Realm.UpdateMode.Modified)
        }
    })
}

export const getManagingTeams = async (id: string): Promise<Team[]> => {
    const realm = await getRealm()
    const teams = await realm.objects<TeamSchema>('Team')
    return teams
        .filter(t => t.managers.map(m => m._id).includes(id))
        .map(t => parseTeam(t))
}

const parseTeam = (schema: TeamSchema): Team => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id,
            place: schema.place,
            name: schema.name,
            teamname: schema.teamname,
            managers: schema.managers,
            players: schema.players,
            seasonStart: schema.seasonStart,
            seasonEnd: schema.seasonEnd,
            seasonNumber: schema.seasonNumber,
            continuationId: schema.continuationId,
            rosterOpen: schema.rosterOpen,
            requests: [],
            games: [],
        }),
    )
}
