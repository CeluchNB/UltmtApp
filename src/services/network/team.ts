import { CreateTeam } from '../../types/team'
import { API_KEY, API_URL_V1 } from '@env'
import axios, { AxiosResponse } from 'axios'

export const createTeam = async (
    token: string,
    data: CreateTeam,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/team`,
        {
            team: { ...data },
        },
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const searchTeam = async (
    term: string,
    rosterOpen?: boolean,
): Promise<AxiosResponse> => {
    return await axios.get(
        `${API_URL_V1}/team/search?q=${term}&rosterOpen=${rosterOpen}`,
        {
            headers: { 'X-API-Key': API_KEY },
        },
    )
}

export const getManagedTeam = async (
    token: string,
    id: string,
): Promise<AxiosResponse> => {
    return await axios.get(`${API_URL_V1}/team/managing/${id}`, {
        headers: {
            Authorization: `Bearer ${token}`,
            'X-API-Key': API_KEY,
        },
    })
}

export const getTeam = async (id: string): Promise<AxiosResponse> => {
    return await axios.get(`${API_URL_V1}/team/${id}`, {
        headers: { 'X-API-Key': API_KEY },
    })
}

export const toggleRosterStatus = async (
    token: string,
    id: string,
    open: boolean,
): Promise<AxiosResponse> => {
    return await axios.put(
        `${API_URL_V1}/team/open/${id}?open=${open}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const removePlayer = async (
    token: string,
    teamId: string,
    userId: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/team/remove/player/${teamId}?user=${userId}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const rollover = async (
    token: string,
    teamId: string,
    copyPlayers: boolean,
    seasonStart: string,
    seasonEnd: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/team/rollover/${teamId}`,
        {
            copyPlayers,
            seasonStart,
            seasonEnd,
        },
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const addManager = async (
    token: string,
    teamId: string,
    managerId: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/team/${teamId}/addManager?manager=${managerId}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}

export const getArchivedTeam = async (
    teamId: string,
): Promise<AxiosResponse> => {
    return await axios.get(`${API_URL_V1}/archiveTeam/${teamId}`, {
        headers: {
            'X-API-Key': API_KEY,
        },
    })
}

export const createBulkJoinCode = async (
    token: string,
    teamId: string,
): Promise<AxiosResponse> => {
    return await axios.post(
        `${API_URL_V1}/team/getBulkCode?id=${teamId}`,
        {},
        {
            headers: {
                Authorization: `Bearer ${token}`,
                'X-API-Key': API_KEY,
            },
        },
    )
}
