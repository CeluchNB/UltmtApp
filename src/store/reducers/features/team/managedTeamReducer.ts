import * as TeamData from '../../../../services/data/team'
import { RootState } from '../../../store'
import { Team } from '../../../../types/team'
import {
    RemovePlayerData,
    TeamRequestData,
    ToggleRosterStatusData,
} from '../../../../types/team'
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit'

export interface ManagedTeamSlice {
    team?: Team
    openLoading: boolean
    teamLoading: boolean
    error?: string
}

const initialState: ManagedTeamSlice = {
    team: undefined,
    openLoading: false,
    teamLoading: false,
    error: undefined,
}

const managedTeamSlice = createSlice({
    name: 'managedTeam',
    initialState,
    reducers: {
        setTeam(state, action) {
            state.team = action.payload
        },
        setTeamLoading(state, action) {
            state.teamLoading = action.payload
        },
    },
    extraReducers: builder => {
        builder
            .addCase(getManagedTeam.pending, state => {
                state.teamLoading = true
                state.error = undefined
            })
            .addCase(getManagedTeam.fulfilled, (state, action) => {
                state.teamLoading = false
                state.error = undefined
                state.team = action.payload
            })
            .addCase(getManagedTeam.rejected, (state, action) => {
                state.teamLoading = false
                state.error = action.error.message
            })
        builder
            .addCase(toggleRosterStatus.pending, state => {
                state.openLoading = true
                state.error = undefined
            })
            .addCase(toggleRosterStatus.fulfilled, (state, action) => {
                state.openLoading = false
                state.error = undefined
                state.team = action.payload
            })
            .addCase(toggleRosterStatus.rejected, (state, action) => {
                state.openLoading = false
                state.error = action.error.message
            })
        builder
            .addCase(removePlayer.pending, state => {
                state.teamLoading = true
                state.error = undefined
            })
            .addCase(removePlayer.fulfilled, (state, action) => {
                state.teamLoading = false
                state.error = undefined
                state.team = action.payload
            })
            .addCase(removePlayer.rejected, (state, action) => {
                state.teamLoading = false
                state.error = action.error.message
            })
    },
})

export const getManagedTeam = createAsyncThunk(
    'managedTeam/getTeam',
    async (data: TeamRequestData, _thunkAPI) => {
        const { token, id } = data
        return await TeamData.getManagedTeam(token, id)
    },
)

export const toggleRosterStatus = createAsyncThunk(
    'managedTeam/toggleRosterStatus',
    async (data: ToggleRosterStatusData, _thunkAPI) => {
        const { token, id, open } = data
        return await TeamData.toggleRosterStatus(token, id, open)
    },
)

export const removePlayer = createAsyncThunk(
    'managedTeam/removePlayer',
    async (data: RemovePlayerData, _thunkAPI) => {
        const { token, id, userId } = data
        return await TeamData.removePlayer(token, id, userId)
    },
)

export const selectTeam = (state: RootState) => state.managedTeam.team
export const selectTeamLoading = (state: RootState) =>
    state.managedTeam.teamLoading
export const selectOpenLoading = (state: RootState) =>
    state.managedTeam.openLoading
export const selectTeamError = (state: RootState) => state.managedTeam.error
export const { setTeam, setTeamLoading } = managedTeamSlice.actions
export default managedTeamSlice.reducer
