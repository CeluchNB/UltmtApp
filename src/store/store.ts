import accountReducer from './reducers/features/account/accountReducer'
import { configureStore } from '@reduxjs/toolkit'
import liveGameReducer from './reducers/features/game/liveGameReducer'
import livePointReducer from './reducers/features/point/livePointReducer'
import managedTeamReducer from './reducers/features/team/managedTeamReducer'

const store = configureStore({
    reducer: {
        account: accountReducer,
        managedTeam: managedTeamReducer,
        liveGame: liveGameReducer,
        livePoint: livePointReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>
export default store
