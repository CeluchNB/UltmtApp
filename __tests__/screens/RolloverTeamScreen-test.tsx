import * as ManagedTeamReducer from '../../src/store/reducers/features/team/managedTeamReducer'
import * as TeamData from '../../src/services/data/team'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import { RolloverTeamProps } from '../../src/types/navigation'
import RolloverTeamScreen from '../../src/screens/RolloverTeamScreen'
import { Team } from '../../src/types/team'
import mockDate from 'mockdate'
import store from '../../src/store/store'
import { act, fireEvent, render } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const goBack = jest.fn()
const navigate = jest.fn()
const setOptions = jest.fn()

const props: RolloverTeamProps = {
    navigation: {
        goBack,
        navigate,
        setOptions,
    } as any,
    route: {} as any,
}

beforeAll(() => {
    store.dispatch(ManagedTeamReducer.setTeam({ _id: 'team1' }))
    mockDate.set('06/01/2022')
})

afterAll(() => {
    mockDate.reset()
})

it('should match snapshot', async () => {
    const snapshot = render(
        <Provider store={store}>
            <NavigationContainer>
                <RolloverTeamScreen {...props} />
            </NavigationContainer>
        </Provider>,
    ).toJSON()

    expect(snapshot).toMatchSnapshot()
})

it('should match snapshot with open requests', async () => {
    const pendingReqProps: RolloverTeamProps = {
        navigation: {
            goBack,
            navigate,
            setOptions,
        } as any,
        route: {} as any,
    }
    const snapshot = render(
        <Provider store={store}>
            <NavigationContainer>
                <RolloverTeamScreen {...pendingReqProps} />
            </NavigationContainer>
        </Provider>,
    ).toJSON()

    expect(snapshot).toMatchSnapshot()
})

it('should handle successful rollover', async () => {
    const spy = jest.spyOn(TeamData, 'rollover').mockReturnValueOnce(
        Promise.resolve({
            _id: 'team1',
        } as Team),
    )
    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <RolloverTeamScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const button = getByText('Submit')
    fireEvent.press(button)
    // Not optimal solution, see
    // ManageTeamDetailsScreen-test.tsx for further details
    await act(async () => {})

    const currentYear = new Date().getFullYear()
    expect(spy).toHaveBeenCalledWith(
        'team1',
        false,
        currentYear.toString(),
        currentYear.toString(),
    )
    expect(navigate).toHaveBeenCalled()
})

it('should handle rollover error', async () => {
    const spy = jest
        .spyOn(TeamData, 'rollover')
        .mockReturnValueOnce(
            Promise.reject({ message: 'Unable to rollover team' }),
        )

    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <RolloverTeamScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const button = getByText('Submit')
    fireEvent.press(button)
    // Not optimal solution, see
    // ManageTeamDetailsScreen-test.tsx for further details
    await act(async () => {})

    expect(spy).toHaveBeenCalled()
    expect(getByText('Unable to rollover team')).toBeTruthy()
})

it('should display custom rollover error', async () => {
    const spy = jest
        .spyOn(TeamData, 'rollover')
        .mockReturnValueOnce(
            Promise.reject({ message: 'custom rollover error' }),
        )

    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <RolloverTeamScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const button = getByText('Submit')
    fireEvent.press(button)
    // Not optimal solution, see
    // ManageTeamDetailsScreen-test.tsx for further details
    await act(async () => {})

    expect(spy).toHaveBeenCalled()
    expect(getByText('custom rollover error')).toBeTruthy()
})
