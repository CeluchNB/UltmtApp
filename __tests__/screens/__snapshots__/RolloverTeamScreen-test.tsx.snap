// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "color": "#ffd600",
          "fontSize": 61.64383561643835,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Start New Season
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "75%",
      }
    }
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "#ffd600",
          "flex": 1,
          "fontSize": 41.0958904109589,
          "fontWeight": "bold",
          "marginTop": 20,
          "width": "75%",
        }
      }
    >
      Keep Current Players
    </Text>
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "checked": false,
          "disabled": false,
        }
      }
      accessibilityValue={
        {
          "text": "off",
        }
      }
      accessible={true}
      pointerEvents="auto"
    >
      <RNCCheckbox
        forwardedRef={null}
        onCheckColor="#ffd600"
        onFillColor="#ffd600"
        onValueChange={[Function]}
        style={
          [
            {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            {
              "alignSelf": "center",
              "marginTop": 25,
            },
          ]
        }
        tintColors={
          {
            "false": "#ababab",
            "true": "#ffd600",
          }
        }
        value={false}
      />
    </View>
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#31adff",
        "marginBottom": 10,
        "width": "75%",
      }
    }
  >
    You can always add and drop players later
  </Text>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#ffd600",
        "fontSize": 41.0958904109589,
        "fontWeight": "bold",
        "marginTop": 20,
        "width": "75%",
      }
    }
  >
    Season
  </Text>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#303030",
        "color": "#ffd600",
        "width": "75%",
      }
    }
  >
    <RNCPicker
      items={
        [
          {
            "label": "2022",
            "testID": undefined,
            "textColor": undefined,
            "value": "2022",
          },
          {
            "label": "2022 - 2023",
            "testID": undefined,
            "textColor": undefined,
            "value": "2022 - 2023",
          },
          {
            "label": "2023",
            "testID": undefined,
            "textColor": undefined,
            "value": "2023",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        [
          {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#31adff",
        "marginBottom": 10,
        "width": "75%",
      }
    }
  >
    Warning: You will not be able to edit the current season after starting the new season.
  </Text>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#31adff",
        "marginBottom": 10,
        "width": "75%",
      }
    }
  >
    Pending requests for this team will all be deleted.
  </Text>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#ffd600",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "minWidth": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    {
                      "marginHorizontal": 8,
                    },
                    false,
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Submit
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should match snapshot with open requests 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "color": "#ffd600",
          "fontSize": 61.64383561643835,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Start New Season
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "75%",
      }
    }
  >
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "#ffd600",
          "flex": 1,
          "fontSize": 41.0958904109589,
          "fontWeight": "bold",
          "marginTop": 20,
          "width": "75%",
        }
      }
    >
      Keep Current Players
    </Text>
    <View
      accessibilityRole="checkbox"
      accessibilityState={
        {
          "checked": false,
          "disabled": false,
        }
      }
      accessibilityValue={
        {
          "text": "off",
        }
      }
      accessible={true}
      pointerEvents="auto"
    >
      <RNCCheckbox
        forwardedRef={null}
        onCheckColor="#ffd600"
        onFillColor="#ffd600"
        onValueChange={[Function]}
        style={
          [
            {
              "backgroundColor": "transparent",
              "height": 32,
              "width": 32,
            },
            {
              "alignSelf": "center",
              "marginTop": 25,
            },
          ]
        }
        tintColors={
          {
            "false": "#ababab",
            "true": "#ffd600",
          }
        }
        value={false}
      />
    </View>
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#31adff",
        "marginBottom": 10,
        "width": "75%",
      }
    }
  >
    You can always add and drop players later
  </Text>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#ffd600",
        "fontSize": 41.0958904109589,
        "fontWeight": "bold",
        "marginTop": 20,
        "width": "75%",
      }
    }
  >
    Season
  </Text>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#303030",
        "color": "#ffd600",
        "width": "75%",
      }
    }
  >
    <RNCPicker
      items={
        [
          {
            "label": "2022",
            "testID": undefined,
            "textColor": undefined,
            "value": "2022",
          },
          {
            "label": "2022 - 2023",
            "testID": undefined,
            "textColor": undefined,
            "value": "2022 - 2023",
          },
          {
            "label": "2023",
            "testID": undefined,
            "textColor": undefined,
            "value": "2023",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        [
          {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#31adff",
        "marginBottom": 10,
        "width": "75%",
      }
    }
  >
    Warning: You will not be able to edit the current season after starting the new season.
  </Text>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#31adff",
        "marginBottom": 10,
        "width": "75%",
      }
    }
  >
    Pending requests for this team will all be deleted.
  </Text>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#ffd600",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "minWidth": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "textAlign": "left",
                  },
                  {
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 24,
                      "marginVertical": 10,
                    },
                    {
                      "marginHorizontal": 8,
                    },
                    false,
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Submit
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
